Upload fails with documents.parsers.ParseError: TypeError: shutdown() got an unexpected keyword argument 'cancel_futures'

[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

<!--- If you are new to Docker or this application our issue tracker is **ONLY** used for reporting bugs or requesting features. Please use [our discord server](https://discord.gg/YWrKVTn) for general support. --->

<!--- Provide a general summary of the bug in the Title above -->

------------------------------

## Expected Behavior
An uploaded image should be stored and processed by OCR

## Current Behavior

The upload crashes on certain images (maybe of a certain size?)

## Steps to Reproduce

1. Capture picture (decent size, e.g. full A4 of text, approx 700KB) in the Android APP 
2. Crop
3. Upload
4. Observe the Logs on the server

## Environment
**OS:** Synology
**CPU architecture:** linux x86_64 
**How docker service was installed:**




## Command used to create docker container (run/create/compose/screenshot)
<!--- Provide your docker create/run command or compose yaml snippet, or a screenshot of settings if using a gui to create the container -->

Docker-compose:

```yaml

version: "3"
services:
  paperless:
    container_name: paperless
    cpu_percent: 25
    mem_limit: 256m
    image: linuxserver/paperless-ngx:latest
    network_mode: "bridge"
    restart: always
    ports:
      - 6688:8000 # host:container
    volumes:
      - ${LOCAL_SOC_LOCATION}/paperless/data:/data
      - ${APP_DATA_LOCATION}/paperless-ngx:/config
    environment:
      - PAPERLESS_DATA_DIR=/config
      - PAPERLESS_MEDIA_ROOT=/data/media
      - PAPERLESS_CONSUMPTION_DIR=/data/consume
 
```

## Docker logs

```bash
[2022-07-08 06:29:09,422] [DEBUG] [paperless.parsing.tesseract] Deleting directory /tmp/paperless/paperless-xlzgcu_k

[2022-07-08 06:29:09,454] [ERROR] [paperless.consumer] Error while consuming document crop_5547.jpeg: TypeError: shutdown() got an unexpected keyword argument 'cancel_futures'

Traceback (most recent call last):

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_exec/tesseract.py", line 226, in get_deskew

    p = run(args_tesseract, stdout=PIPE, stderr=STDOUT, timeout=timeout, check=True)

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/subprocess/__init__.py", line 53, in run

    proc = subprocess_run(args, env=env, **kwargs)

  File "/usr/lib/python3.8/subprocess.py", line 516, in run

    raise CalledProcessError(retcode, process.args,

subprocess.CalledProcessError: Command '['tesseract', '-l', 'eng', '--psm', '2', '/tmp/ocrmypdf.io._d8p80vg/000001_rasterize.png', 'stdout']' died with <Signals.SIGKILL: 9>.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/builtin_plugins/concurrency.py", line 135, in _execute

    result = future.result()

  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 437, in result

    return self.__get_result()

  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result

    raise self._exception

  File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run

    result = self.fn(*self.args, **self.kwargs)

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_sync.py", line 192, in exec_page_sync

    ocr_image, preprocess_out = make_intermediate_images(

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_sync.py", line 135, in make_intermediate_images

    preprocess_out = preprocess(

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_sync.py", line 106, in preprocess

    image = preprocess_deskew(image, page_context)

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_pipeline.py", line 488, in preprocess_deskew

    deskew_angle_degrees = ocr_engine.get_deskew(input_file, page_context.options)

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/builtin_plugins/tesseract_ocr.py", line 167, in get_deskew

    return tesseract.get_deskew(

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_exec/tesseract.py", line 237, in get_deskew

    raise SubprocessOutputError() from e

ocrmypdf.exceptions.SubprocessOutputError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "/app/paperless/src/paperless_tesseract/parsers.py", line 261, in parse

    ocrmypdf.ocr(**args)

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/api.py", line 337, in ocr

    return run_pipeline(options=options, plugin_manager=plugin_manager, api=True)

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_sync.py", line 385, in run_pipeline

    exec_concurrent(context, executor)

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_sync.py", line 274, in exec_concurrent

    executor(

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_concurrent.py", line 82, in __call__

    self._execute(

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/builtin_plugins/concurrency.py", line 148, in _execute

    executor.shutdown(wait=False, cancel_futures=True)

TypeError: shutdown() got an unexpected keyword argument 'cancel_futures'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "/app/paperless/src/documents/consumer.py", line 261, in try_consume_file

    document_parser.parse(self.path, mime_type, self.filename)

  File "/app/paperless/src/paperless_tesseract/parsers.py", line 314, in parse

    raise ParseError(f"{e.__class__.__name__}: {str(e)}")

documents.parsers.ParseError: TypeError: shutdown() got an unexpected keyword argument 'cancel_futures'
```

I assume this issue is related to an upgrade of Python to 3.8. A similar issue was reported on this project here: https://github.com/DiscordTime/ggdrive/issues/56

Can confirm that behaviour on my side. 

```
[2022-07-10 13:11:33,318] [ERROR] [paperless.consumer] Error while consuming document dhvinfo201.pdf: TypeError: shutdown() got an unexpected keyword argument 'cancel_futures'
paperless-ngx    | Traceback (most recent call last):
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_exec/ghostscript.py", line 126, in rasterize_pdf
paperless-ngx    |     p = run(args_gs, stdout=PIPE, stderr=PIPE, check=True)
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/subprocess/__init__.py", line 53, in run
paperless-ngx    |     proc = subprocess_run(args, env=env, **kwargs)
paperless-ngx    |   File "/usr/lib/python3.8/subprocess.py", line 516, in run
paperless-ngx    |     raise CalledProcessError(retcode, process.args,
paperless-ngx    | subprocess.CalledProcessError: Command '['gs', '-dQUIET', '-dSAFER', '-dBATCH', '-dNOPAUSE', '-dInterpolateControl=-1', '-sDEVICE=jpeggray', '-dFirstPage=55', '-dLastPage=55', '-r9174.124890x9174.124890', '-o', '-', '-sstdout=%stderr', '-dAutoRotatePages=/None', '-f', '/tmp/ocrmypdf.io._l9rqjsy/origin.pdf']' returned non-zero exit status 1.
paperless-ngx    |
paperless-ngx    | During handling of the above exception, another exception occurred:
paperless-ngx    |
paperless-ngx    | Traceback (most recent call last):
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/builtin_plugins/concurrency.py", line 135, in _execute
paperless-ngx    |     result = future.result()
paperless-ngx    |   File "/usr/lib/python3.8/concurrent/futures/_base.py", line 437, in result
paperless-ngx    |     return self.__get_result()
paperless-ngx    |   File "/usr/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
paperless-ngx    |     raise self._exception
paperless-ngx    |   File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
paperless-ngx    |     result = self.fn(*self.args, **self.kwargs)
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_sync.py", line 187, in exec_page_sync
paperless-ngx    |     rasterize_preview_out = rasterize_preview(page_context.origin, page_context)
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_pipeline.py", line 344, in rasterize_preview
paperless-ngx    |     page_context.plugin_manager.hook.rasterize_pdf_page(
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/pluggy/_hooks.py", line 265, in __call__
paperless-ngx    |     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/pluggy/_manager.py", line 80, in _hookexec
paperless-ngx    |     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/pluggy/_callers.py", line 60, in _multicall
paperless-ngx    |     return outcome.get_result()
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/pluggy/_result.py", line 60, in get_result
paperless-ngx    |     raise ex[1].with_traceback(ex[2])
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/pluggy/_callers.py", line 39, in _multicall
paperless-ngx    |     res = hook_impl.function(*args)
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/builtin_plugins/ghostscript.py", line 68, in rasterize_pdf_page
paperless-ngx    |     ghostscript.rasterize_pdf(
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_exec/ghostscript.py", line 129, in rasterize_pdf
paperless-ngx    |     raise SubprocessOutputError('Ghostscript rasterizing failed')
paperless-ngx    | ocrmypdf.exceptions.SubprocessOutputError: Ghostscript rasterizing failed
paperless-ngx    |
paperless-ngx    | During handling of the above exception, another exception occurred:
paperless-ngx    |
paperless-ngx    | Traceback (most recent call last):
paperless-ngx    |   File "/app/paperless/src/paperless_tesseract/parsers.py", line 261, in parse
paperless-ngx    |     ocrmypdf.ocr(**args)
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/api.py", line 337, in ocr
paperless-ngx    |     return run_pipeline(options=options, plugin_manager=plugin_manager, api=True)
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_sync.py", line 385, in run_pipeline
paperless-ngx    |     exec_concurrent(context, executor)
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_sync.py", line 274, in exec_concurrent
paperless-ngx    |     executor(
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_concurrent.py", line 82, in __call__
paperless-ngx    |     self._execute(
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/builtin_plugins/concurrency.py", line 148, in _execute
paperless-ngx    |     executor.shutdown(wait=False, cancel_futures=True)
paperless-ngx    | TypeError: shutdown() got an unexpected keyword argument 'cancel_futures'
paperless-ngx    |
paperless-ngx    | During handling of the above exception, another exception occurred:
paperless-ngx    |
paperless-ngx    | Traceback (most recent call last):
paperless-ngx    |   File "/app/paperless/src/documents/consumer.py", line 261, in try_consume_file
paperless-ngx    |     document_parser.parse(self.path, mime_type, self.filename)
paperless-ngx    |   File "/app/paperless/src/paperless_tesseract/parsers.py", line 314, in parse
paperless-ngx    |     raise ParseError(f"{e.__class__.__name__}: {str(e)}")
paperless-ngx    | documents.parsers.ParseError: TypeError: shutdown() got an unexpected keyword argument 'cancel_futures'
paperless-ngx    | 13:11:33 [Q] INFO Process-1:7 stopped doing work
paperless-ngx    | 13:11:33 [Q] ERROR Failed [dhvinfo201.pdf] - dhvinfo201.pdf: Error while consuming document dhvinfo201.pdf: TypeError: shutdown() got an unexpected keyword argument 'cancel_futures' : Traceback (most recent call last):
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_exec/ghostscript.py", line 126, in rasterize_pdf
paperless-ngx    |     p = run(args_gs, stdout=PIPE, stderr=PIPE, check=True)
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/subprocess/__init__.py", line 53, in run
paperless-ngx    |     proc = subprocess_run(args, env=env, **kwargs)
paperless-ngx    |   File "/usr/lib/python3.8/subprocess.py", line 516, in run
paperless-ngx    |     raise CalledProcessError(retcode, process.args,
paperless-ngx    | subprocess.CalledProcessError: Command '['gs', '-dQUIET', '-dSAFER', '-dBATCH', '-dNOPAUSE', '-dInterpolateControl=-1', '-sDEVICE=jpeggray', '-dFirstPage=55', '-dLastPage=55', '-r9174.124890x9174.124890', '-o', '-', '-sstdout=%stderr', '-dAutoRotatePages=/None', '-f', '/tmp/ocrmypdf.io._l9rqjsy/origin.pdf']' returned non-zero exit status 1.
paperless-ngx    |
paperless-ngx    | During handling of the above exception, another exception occurred:
paperless-ngx    |
paperless-ngx    | Traceback (most recent call last):
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/builtin_plugins/concurrency.py", line 135, in _execute
paperless-ngx    |     result = future.result()
paperless-ngx    |   File "/usr/lib/python3.8/concurrent/futures/_base.py", line 437, in result
paperless-ngx    |     return self.__get_result()
paperless-ngx    |   File "/usr/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
paperless-ngx    |     raise self._exception
paperless-ngx    |   File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
paperless-ngx    |     result = self.fn(*self.args, **self.kwargs)
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_sync.py", line 187, in exec_page_sync
paperless-ngx    |     rasterize_preview_out = rasterize_preview(page_context.origin, page_context)
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_pipeline.py", line 344, in rasterize_preview
paperless-ngx    |     page_context.plugin_manager.hook.rasterize_pdf_page(
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/pluggy/_hooks.py", line 265, in __call__
paperless-ngx    |     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/pluggy/_manager.py", line 80, in _hookexec
paperless-ngx    |     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/pluggy/_callers.py", line 60, in _multicall
paperless-ngx    |     return outcome.get_result()
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/pluggy/_result.py", line 60, in get_result
paperless-ngx    |     raise ex[1].with_traceback(ex[2])
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/pluggy/_callers.py", line 39, in _multicall
paperless-ngx    |     res = hook_impl.function(*args)
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/builtin_plugins/ghostscript.py", line 68, in rasterize_pdf_page
paperless-ngx    |     ghostscript.rasterize_pdf(
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_exec/ghostscript.py", line 129, in rasterize_pdf
paperless-ngx    |     raise SubprocessOutputError('Ghostscript rasterizing failed')
paperless-ngx    | ocrmypdf.exceptions.SubprocessOutputError: Ghostscript rasterizing failed
paperless-ngx    |
paperless-ngx    | During handling of the above exception, another exception occurred:
paperless-ngx    |
paperless-ngx    | Traceback (most recent call last):
paperless-ngx    |   File "/app/paperless/src/paperless_tesseract/parsers.py", line 261, in parse
paperless-ngx    |     ocrmypdf.ocr(**args)
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/api.py", line 337, in ocr
paperless-ngx    |     return run_pipeline(options=options, plugin_manager=plugin_manager, api=True)
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_sync.py", line 385, in run_pipeline
paperless-ngx    |     exec_concurrent(context, executor)
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_sync.py", line 274, in exec_concurrent
paperless-ngx    |     executor(
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_concurrent.py", line 82, in __call__
paperless-ngx    |     self._execute(
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/builtin_plugins/concurrency.py", line 148, in _execute
paperless-ngx    |     executor.shutdown(wait=False, cancel_futures=True)
paperless-ngx    | TypeError: shutdown() got an unexpected keyword argument 'cancel_futures'
paperless-ngx    |
paperless-ngx    | During handling of the above exception, another exception occurred:
paperless-ngx    |
paperless-ngx    | Traceback (most recent call last):
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/asgiref/sync.py", line 275, in main_wrap
paperless-ngx    |     raise exc_info[1]
paperless-ngx    |   File "/app/paperless/src/documents/consumer.py", line 261, in try_consume_file
paperless-ngx    |     document_parser.parse(self.path, mime_type, self.filename)
paperless-ngx    |   File "/app/paperless/src/paperless_tesseract/parsers.py", line 314, in parse
paperless-ngx    |     raise ParseError(f"{e.__class__.__name__}: {str(e)}")
paperless-ngx    | documents.parsers.ParseError: TypeError: shutdown() got an unexpected keyword argument 'cancel_futures'
paperless-ngx    |
paperless-ngx    | During handling of the above exception, another exception occurred:
paperless-ngx    |
paperless-ngx    | Traceback (most recent call last):
paperless-ngx    |   File "/usr/local/lib/python3.8/dist-packages/django_q/cluster.py", line 432, in worker
paperless-ngx    |     res = f(*task["args"], **task["kwargs"])
paperless-ngx    |   File "/app/paperless/src/documents/tasks.py", line 298, in consume_file
paperless-ngx    |     document = Consumer().try_consume_file(
paperless-ngx    |   File "/app/paperless/src/documents/consumer.py", line 280, in try_consume_file
paperless-ngx    |     self._fail(
paperless-ngx    |   File "/app/paperless/src/documents/consumer.py", line 81, in _fail
paperless-ngx    |     raise ConsumerError(f"{self.filename}: {log_message or message}")
paperless-ngx    | documents.consumer.ConsumerError: dhvinfo201.pdf: Error while consuming document dhvinfo201.pdf: TypeError: shutdown() got an unexpected keyword argument 'cancel_futures'
paperless-ngx    |
paperless-ngx    | 13:11:34 [Q] INFO recycled worker Process-1:7
```

With the upstream image the same file causes an exception as well. However, the messages are a bit differen. Rasterizing failed is the one message that still is there. That `cancel_futures` stuff is not present as it is in the lsio image. 

So maybe the python 3.9 upgrade changed things but not everything.

```
webserver_1  | [2022-07-13 16:07:54,948] [INFO] [paperless.management.consumer] Adding /usr/src/paperless/src/../consume/dhvinfo201.pdf to the task queue.
webserver_1  | 16:07:54 [Q] INFO Enqueued 1
webserver_1  | 16:07:54 [Q] INFO Process-1:9 processing [dhvinfo201.pdf]
webserver_1  | [2022-07-13 16:07:55,387] [INFO] [paperless.consumer] Consuming dhvinfo201.pdf
webserver_1  | 16:16:35 [Q] INFO Enqueued 1
webserver_1  | 16:16:35 [Q] INFO Process-1:10 processing [magnesium-hot-moon-lemon]
webserver_1  | 16:16:35 [Q] INFO Process-1 created a task from schedule [Check all e-mail accounts]
webserver_1  | 16:16:35 [Q] INFO Process-1:10 stopped doing work
webserver_1  | 16:16:35 [Q] INFO Processed [magnesium-hot-moon-lemon]
webserver_1  | 16:16:36 [Q] INFO recycled worker Process-1:10
webserver_1  | 16:16:36 [Q] INFO Process-1:11 ready for work at 708
webserver_1  | [2022-07-13 16:22:16,423] [WARNING] [ocrmypdf._exec.tesseract] [tesseract] lots of diacritics - possibly poor OCR
webserver_1  | [2022-07-13 16:23:13,127] [WARNING] [ocrmypdf._exec.tesseract] [tesseract] lots of diacritics - possibly poor OCR
webserver_1  | [2022-07-13 16:23:15,588] [ERROR] [ocrmypdf._exec.ghostscript] Error: /ioerror in --showpage--
webserver_1  | Operand stack:
webserver_1  |    1   true
webserver_1  | Execution stack:
webserver_1  |    %interp_exit   .runexec2   --nostringval--   showpage   --nostringval--   2   %stopped_push   --nostringval--   showpage   showpage   false   1   %stopped_push   1990   1   3   %oparray_pop   1989   1   3   %oparray_pop   1977   1   3   %oparray_pop   1978   1   3   %oparray_pop   showpage   showpage   56   1   55   showpage   %for_pos_int_continue   1981   1   7   %oparray_pop   showpage   showpage   1840   0   9   %oparray_pop   showpage   showpage
webserver_1  | Dictionary stack:
webserver_1  |    --dict:738/1123(ro)(G)--   --dict:1/20(G)--   --dict:80/200(L)--   --dict:80/200(L)--   --dict:133/256(ro)(G)--   --dict:320/325(ro)(G)--   --dict:31/32(L)--   --dict:6/9(L)--   --dict:7/20(L)--
webserver_1  | Current allocation mode is local
webserver_1  | GPL Ghostscript 9.53.3: Unrecoverable error, exit code 1
webserver_1  |
webserver_1  | [2022-07-13 16:23:15,848] [ERROR] [paperless.consumer] Error while consuming document dhvinfo201.pdf: SubprocessOutputError: Ghostscript rasterizing failed
webserver_1  | Traceback (most recent call last):
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/_exec/ghostscript.py", line 126, in rasterize_pdf
webserver_1  |     p = run(args_gs, stdout=PIPE, stderr=PIPE, check=True)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/subprocess/__init__.py", line 53, in run
webserver_1  |     proc = subprocess_run(args, env=env, **kwargs)
webserver_1  |   File "/usr/local/lib/python3.9/subprocess.py", line 528, in run
webserver_1  |     raise CalledProcessError(retcode, process.args,
webserver_1  | subprocess.CalledProcessError: Command '['gs', '-dQUIET', '-dSAFER', '-dBATCH', '-dNOPAUSE', '-dInterpolateControl=-1', '-sDEVICE=jpeggray', '-dFirstPage=55', '-dLastPage=55', '-r9174.124890x9174.124890', '-o', '-', '-sstdout=%stderr', '-dAutoRotatePages=/None', '-f', '/tmp/ocrmypdf.io.ptvzbqoo/origin.pdf']' returned non-zero exit status 1.
webserver_1  |
webserver_1  | During handling of the above exception, another exception occurred:
webserver_1  |
webserver_1  | Traceback (most recent call last):
webserver_1  |   File "/usr/src/paperless/src/paperless_tesseract/parsers.py", line 261, in parse
webserver_1  |     ocrmypdf.ocr(**args)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/api.py", line 337, in ocr
webserver_1  |     return run_pipeline(options=options, plugin_manager=plugin_manager, api=True)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/_sync.py", line 385, in run_pipeline
webserver_1  |     exec_concurrent(context, executor)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/_sync.py", line 274, in exec_concurrent
webserver_1  |     executor(
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/_concurrent.py", line 82, in __call__
webserver_1  |     self._execute(
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/builtin_plugins/concurrency.py", line 135, in _execute
webserver_1  |     result = future.result()
webserver_1  |   File "/usr/local/lib/python3.9/concurrent/futures/_base.py", line 439, in result
webserver_1  |     return self.__get_result()
webserver_1  |   File "/usr/local/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
webserver_1  |     raise self._exception
webserver_1  |   File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
webserver_1  |     result = self.fn(*self.args, **self.kwargs)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/_sync.py", line 187, in exec_page_sync
webserver_1  |     rasterize_preview_out = rasterize_preview(page_context.origin, page_context)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/_pipeline.py", line 344, in rasterize_preview
webserver_1  |     page_context.plugin_manager.hook.rasterize_pdf_page(
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/pluggy/_hooks.py", line 265, in __call__
webserver_1  |     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/pluggy/_manager.py", line 80, in _hookexec
webserver_1  |     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/pluggy/_callers.py", line 60, in _multicall
webserver_1  |     return outcome.get_result()
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/pluggy/_result.py", line 60, in get_result
webserver_1  |     raise ex[1].with_traceback(ex[2])
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/pluggy/_callers.py", line 39, in _multicall
webserver_1  |     res = hook_impl.function(*args)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/builtin_plugins/ghostscript.py", line 68, in rasterize_pdf_page
webserver_1  |     ghostscript.rasterize_pdf(
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/_exec/ghostscript.py", line 129, in rasterize_pdf
webserver_1  |     raise SubprocessOutputError('Ghostscript rasterizing failed')
webserver_1  | ocrmypdf.exceptions.SubprocessOutputError: Ghostscript rasterizing failed
webserver_1  |
webserver_1  | During handling of the above exception, another exception occurred:
webserver_1  |
webserver_1  | Traceback (most recent call last):
webserver_1  |   File "/usr/src/paperless/src/documents/consumer.py", line 261, in try_consume_file
webserver_1  |     document_parser.parse(self.path, mime_type, self.filename)
webserver_1  |   File "/usr/src/paperless/src/paperless_tesseract/parsers.py", line 314, in parse
webserver_1  |     raise ParseError(f"{e.__class__.__name__}: {str(e)}")
webserver_1  | documents.parsers.ParseError: SubprocessOutputError: Ghostscript rasterizing failed
webserver_1  | 16:23:15 [Q] INFO Process-1:9 stopped doing work
webserver_1  | 16:23:16 [Q] ERROR Failed [dhvinfo201.pdf] - dhvinfo201.pdf: Error while consuming document dhvinfo201.pdf: SubprocessOutputError: Ghostscript rasterizing failed : Traceback (most recent call last):
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/_exec/ghostscript.py", line 126, in rasterize_pdf
webserver_1  |     p = run(args_gs, stdout=PIPE, stderr=PIPE, check=True)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/subprocess/__init__.py", line 53, in run
webserver_1  |     proc = subprocess_run(args, env=env, **kwargs)
webserver_1  |   File "/usr/local/lib/python3.9/subprocess.py", line 528, in run
webserver_1  |     raise CalledProcessError(retcode, process.args,
webserver_1  | subprocess.CalledProcessError: Command '['gs', '-dQUIET', '-dSAFER', '-dBATCH', '-dNOPAUSE', '-dInterpolateControl=-1', '-sDEVICE=jpeggray', '-dFirstPage=55', '-dLastPage=55', '-r9174.124890x9174.124890', '-o', '-', '-sstdout=%stderr', '-dAutoRotatePages=/None', '-f', '/tmp/ocrmypdf.io.ptvzbqoo/origin.pdf']' returned non-zero exit status 1.
webserver_1  |
webserver_1  | During handling of the above exception, another exception occurred:
webserver_1  |
webserver_1  | Traceback (most recent call last):
webserver_1  |   File "/usr/src/paperless/src/paperless_tesseract/parsers.py", line 261, in parse
webserver_1  |     ocrmypdf.ocr(**args)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/api.py", line 337, in ocr
webserver_1  |     return run_pipeline(options=options, plugin_manager=plugin_manager, api=True)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/_sync.py", line 385, in run_pipeline
webserver_1  |     exec_concurrent(context, executor)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/_sync.py", line 274, in exec_concurrent
webserver_1  |     executor(
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/_concurrent.py", line 82, in __call__
webserver_1  |     self._execute(
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/builtin_plugins/concurrency.py", line 135, in _execute
webserver_1  |     result = future.result()
webserver_1  |   File "/usr/local/lib/python3.9/concurrent/futures/_base.py", line 439, in result
webserver_1  |     return self.__get_result()
webserver_1  |   File "/usr/local/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
webserver_1  |     raise self._exception
webserver_1  |   File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
webserver_1  |     result = self.fn(*self.args, **self.kwargs)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/_sync.py", line 187, in exec_page_sync
webserver_1  |     rasterize_preview_out = rasterize_preview(page_context.origin, page_context)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/_pipeline.py", line 344, in rasterize_preview
webserver_1  |     page_context.plugin_manager.hook.rasterize_pdf_page(
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/pluggy/_hooks.py", line 265, in __call__
webserver_1  |     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/pluggy/_manager.py", line 80, in _hookexec
webserver_1  |     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/pluggy/_callers.py", line 60, in _multicall
webserver_1  |     return outcome.get_result()
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/pluggy/_result.py", line 60, in get_result
webserver_1  |     raise ex[1].with_traceback(ex[2])
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/pluggy/_callers.py", line 39, in _multicall
webserver_1  |     res = hook_impl.function(*args)
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/builtin_plugins/ghostscript.py", line 68, in rasterize_pdf_page
webserver_1  |     ghostscript.rasterize_pdf(
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/ocrmypdf/_exec/ghostscript.py", line 129, in rasterize_pdf
webserver_1  |     raise SubprocessOutputError('Ghostscript rasterizing failed')
webserver_1  | ocrmypdf.exceptions.SubprocessOutputError: Ghostscript rasterizing failed
webserver_1  |
webserver_1  | During handling of the above exception, another exception occurred:
webserver_1  |
webserver_1  | Traceback (most recent call last):
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/asgiref/sync.py", line 275, in main_wrap
webserver_1  |     raise exc_info[1]
webserver_1  |   File "/usr/src/paperless/src/documents/consumer.py", line 261, in try_consume_file
webserver_1  |     document_parser.parse(self.path, mime_type, self.filename)
webserver_1  |   File "/usr/src/paperless/src/paperless_tesseract/parsers.py", line 314, in parse
webserver_1  |     raise ParseError(f"{e.__class__.__name__}: {str(e)}")
webserver_1  | documents.parsers.ParseError: SubprocessOutputError: Ghostscript rasterizing failed
webserver_1  |
webserver_1  | During handling of the above exception, another exception occurred:
webserver_1  |
webserver_1  | Traceback (most recent call last):
webserver_1  |   File "/usr/local/lib/python3.9/site-packages/django_q/cluster.py", line 432, in worker
webserver_1  |     res = f(*task["args"], **task["kwargs"])
webserver_1  |   File "/usr/src/paperless/src/documents/tasks.py", line 298, in consume_file
webserver_1  |     document = Consumer().try_consume_file(
webserver_1  |   File "/usr/src/paperless/src/documents/consumer.py", line 280, in try_consume_file
webserver_1  |     self._fail(
webserver_1  |   File "/usr/src/paperless/src/documents/consumer.py", line 81, in _fail
webserver_1  |     raise ConsumerError(f"{self.filename}: {log_message or message}")
webserver_1  | documents.consumer.ConsumerError: dhvinfo201.pdf: Error while consuming document dhvinfo201.pdf: SubprocessOutputError: Ghostscript rasterizing failed
webserver_1  |
webserver_1  | 16:23:16 [Q] INFO recycled worker Process-1:9

```

Hey, tried the image `lspipepr/paperless-ngx:v1.7.1-pkg-92a7a783-pr-31` with the file causing previous errors and getting this:

```bash

[2022-07-17 00:59:59,566] [INFO] [paperless.management.consumer] Using inotify to watch directory for changes: /data/consume

[2022-07-17 01:00:31,171] [INFO] [paperless.sanity_checker] Sanity checker detected no issues.

[2022-07-17 01:01:05,212] [INFO] [paperless.consumer] Consuming 0000048.jpg

[2022-07-17 01:01:05,218] [DEBUG] [paperless.consumer] Detected mime type: image/jpeg

[2022-07-17 01:01:05,225] [DEBUG] [paperless.consumer] Parser: RasterisedDocumentParser

[2022-07-17 01:01:05,229] [DEBUG] [paperless.consumer] Parsing 0000048.jpg...

[2022-07-17 01:01:05,514] [DEBUG] [paperless.parsing.tesseract] Estimated DPI 328 based on image width 2720

[2022-07-17 01:01:05,515] [DEBUG] [paperless.parsing.tesseract] Detected DPI for image /tmp/paperless/paperless-upload-312hy62s: 72

[2022-07-17 01:01:05,515] [DEBUG] [paperless.parsing.tesseract] Calling OCRmyPDF with args: {'input_file': '/tmp/paperless/paperless-upload-312hy62s', 'output_file': '/tmp/paperless/paperless-y505ggmj/archive.pdf', 'use_threads': True, 'jobs': 2, 'language': 'eng', 'output_type': 'pdfa', 'progress_bar': False, 'skip_text': True, 'clean': True, 'deskew': True, 'rotate_pages': True, 'rotate_pages_threshold': 12.0, 'sidecar': '/tmp/paperless/paperless-y505ggmj/sidecar.txt', 'image_dpi': 72}

[2022-07-17 01:01:19,810] [DEBUG] [paperless.parsing.tesseract] Deleting directory /tmp/paperless/paperless-y505ggmj

[2022-07-17 01:01:19,843] [ERROR] [paperless.consumer] Error while consuming document 0000048.jpg: SubprocessOutputError:

Traceback (most recent call last):

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/_exec/tesseract.py", line 226, in get_deskew

    p = run(args_tesseract, stdout=PIPE, stderr=STDOUT, timeout=timeout, check=True)

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/subprocess/__init__.py", line 53, in run

    proc = subprocess_run(args, env=env, **kwargs)

  File "/usr/lib/python3.10/subprocess.py", line 524, in run

    raise CalledProcessError(retcode, process.args,

subprocess.CalledProcessError: Command '['tesseract', '-l', 'eng', '--psm', '2', '/tmp/ocrmypdf.io.qujac1y1/000001_rasterize.png', 'stdout']' died with <Signals.SIGKILL: 9>.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):

  File "/app/paperless/src/paperless_tesseract/parsers.py", line 261, in parse

    ocrmypdf.ocr(**args)

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/api.py", line 337, in ocr

    return run_pipeline(options=options, plugin_manager=plugin_manager, api=True)

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/_sync.py", line 385, in run_pipeline

    exec_concurrent(context, executor)

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/_sync.py", line 274, in exec_concurrent

    executor(

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/_concurrent.py", line 82, in __call__

    self._execute(

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/builtin_plugins/concurrency.py", line 135, in _execute

    result = future.result()

  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 439, in result

    return self.__get_result()

  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 391, in __get_result

    raise self._exception

  File "/usr/lib/python3.10/concurrent/futures/thread.py", line 58, in run

    result = self.fn(*self.args, **self.kwargs)

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/_sync.py", line 192, in exec_page_sync

    ocr_image, preprocess_out = make_intermediate_images(

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/_sync.py", line 135, in make_intermediate_images

    preprocess_out = preprocess(

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/_sync.py", line 106, in preprocess

    image = preprocess_deskew(image, page_context)

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/_pipeline.py", line 488, in preprocess_deskew

    deskew_angle_degrees = ocr_engine.get_deskew(input_file, page_context.options)

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/builtin_plugins/tesseract_ocr.py", line 167, in get_deskew

    return tesseract.get_deskew(

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/_exec/tesseract.py", line 237, in get_deskew

    raise SubprocessOutputError() from e

ocrmypdf.exceptions.SubprocessOutputError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "/app/paperless/src/documents/consumer.py", line 261, in try_consume_file

    document_parser.parse(self.path, mime_type, self.filename)

  File "/app/paperless/src/paperless_tesseract/parsers.py", line 314, in parse

    raise ParseError(f"{e.__class__.__name__}: {str(e)}")

documents.parsers.ParseError: SubprocessOutputError:

```

A random other file:

```bash

[2022-07-17 01:04:47,029] [DEBUG] [paperless.parsing.tesseract] Deleting directory /tmp/paperless/paperless-c5_822lj

[2022-07-17 01:04:47,061] [ERROR] [paperless.consumer] Error while consuming document 0000049.jpg: CalledProcessError: Command '['unpaper', '-v', '--dpi', '130.000003', '--layout', 'none', '--mask-scan-size', '100', '--no-border-align', '--no-mask-center', '--no-grayfilter', '--no-blackfilter', '--no-deskew', '/tmp/tmpa46sqnp0/input.pnm', '/tmp/tmpa46sqnp0/output.ppm']' died with <Signals.SIGKILL: 9>.

Traceback (most recent call last):

  File "/app/paperless/src/paperless_tesseract/parsers.py", line 261, in parse

    ocrmypdf.ocr(**args)

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/api.py", line 337, in ocr

    return run_pipeline(options=options, plugin_manager=plugin_manager, api=True)

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/_sync.py", line 385, in run_pipeline

    exec_concurrent(context, executor)

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/_sync.py", line 274, in exec_concurrent

    executor(

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/_concurrent.py", line 82, in __call__

    self._execute(

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/builtin_plugins/concurrency.py", line 135, in _execute

    result = future.result()

  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 439, in result

    return self.__get_result()

  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 391, in __get_result

    raise self._exception

  File "/usr/lib/python3.10/concurrent/futures/thread.py", line 58, in run

    result = self.fn(*self.args, **self.kwargs)

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/_sync.py", line 192, in exec_page_sync

    ocr_image, preprocess_out = make_intermediate_images(

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/_sync.py", line 161, in make_intermediate_images

    ocr_image = preprocess(

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/_sync.py", line 108, in preprocess

    image = preprocess_clean(image, page_context)

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/_pipeline.py", line 506, in preprocess_clean

    return unpaper.clean(

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/_exec/unpaper.py", line 160, in clean

    run_unpaper(input_file, output_file, dpi=dpi, mode_args=unpaper_args)

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/_exec/unpaper.py", line 112, in run_unpaper

    run(

  File "/usr/local/lib/python3.10/dist-packages/ocrmypdf/subprocess/__init__.py", line 53, in run

    proc = subprocess_run(args, env=env, **kwargs)

  File "/usr/lib/python3.10/subprocess.py", line 524, in run

    raise CalledProcessError(retcode, process.args,

subprocess.CalledProcessError: Command '['unpaper', '-v', '--dpi', '130.000003', '--layout', 'none', '--mask-scan-size', '100', '--no-border-align', '--no-mask-center', '--no-grayfilter', '--no-blackfilter', '--no-deskew', '/tmp/tmpa46sqnp0/input.pnm', '/tmp/tmpa46sqnp0/output.ppm']' died with <Signals.SIGKILL: 9>.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "/app/paperless/src/documents/consumer.py", line 261, in try_consume_file

    document_parser.parse(self.path, mime_type, self.filename)

  File "/app/paperless/src/paperless_tesseract/parsers.py", line 314, in parse

    raise ParseError(f"{e.__class__.__name__}: {str(e)}")

documents.parsers.ParseError: CalledProcessError: Command '['unpaper', '-v', '--dpi', '130.000003', '--layout', 'none', '--mask-scan-size', '100', '--no-border-align', '--no-mask-center', '--no-grayfilter', '--no-blackfilter', '--no-deskew', '/tmp/tmpa46sqnp0/input.pnm', '/tmp/tmpa46sqnp0/output.ppm']' died with <Signals.SIGKILL: 9>.
```

--- a/readme-vars.yml
+++ b/readme-vars.yml
@@ -48,6 +48,7 @@ app_setup_block: |
   For convenience this container provides an alias to perform administration management commands. Available administration commands are documented upstream [here](https://paperless-ng.readthedocs.io/en/latest/administration.html) and can be accessed with this container thus: `docker exec -it <container_name> manage <command>`. For example, `docker exec -it paperless manage  document_retagger -tT`.
 # changelog
 changelogs:
+  - { date: "05.09.22:", desc: "Deprecate." }
   - { date: "16.05.22:", desc: "Add correct libqpdf.so to arm builds." }
   - { date: "14.05.22:", desc: "Fine tune disabling of redis." }
   - { date: "12.05.22:", desc: "Move migrations to after multilangocr mod. Fix disabling of redis. Add missing dep for postgresql." }

--- a/readme-vars.yml
+++ b/readme-vars.yml
@@ -6,8 +6,12 @@ project_url: "https://github.com/paperless-ngx/paperless-ngx"
 project_logo: "https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/paperless-ngx-banner.png"
 project_blurb: |
   [{{ project_name|capitalize }}]({{ project_url }}) is an application by Daniel Quinn and contributors that indexes your scanned documents and allows you to easily search for documents and store metadata alongside your documents."
-
 project_lsio_github_repo_url: "https://github.com/linuxserver/docker-{{ project_name }}"
+
+project_deprecation_status: true
+project_deprecation_message: |
+  We recommend switching to the new official container: 
+  https://github.com/paperless-ngx/paperless-ngx
 
 # supported architectures
 available_architectures:

